### PR TITLE
Bug 1170067 - [Stingray][EPG] Make program element as a web component

### PR DESCRIPTION
**Pull request by sean2449**
- Make program element as a web component in epg_program
- Move out related css
- Add test cases

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=13a0bde1a3f2fd3bd79dfe649386404858b9d64b

File: tv_apps/tv-epg/js/epg_program.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    scaleX = scaleX / (this.duration * EPG.COLUMN_WIDTH - EPG.COLUMN_MARGIN);
+    scaleX = Math.min(scaleX, 1);
+    this.progressElement.style.transform = 'scaleX(' + scaleX + ')';
+  };
+
+  proto.fillProgress = function() {
+    this.progressElement.style.transform = 'scaleX(1)';
+  };
+
+  proto.setStartTime = function(time) {
```
Comment (john-hu) [inline tv_apps/tv-epg/js/epg_program.js]: How about to use property getter and setter for these parts?

`startTime, title, titlePadding (maybe), duration`

File: tv_apps/tv-epg/js/epg.js [hunk 2]
```diff
@@ -263,8 +234,7 @@
       if (prevPrograms && prevPrograms[row] &&
           programElement !== prevPrograms[row].element) {
         programElement = prevPrograms[row].element;
-        programElement.querySelector('.background-progress').style.transform =
-          'scaleX(1)';
+        programElement.fillProgress();
```
Comment (john-hu) [inline tv_apps/tv-epg/js/epg.js]: May we fill progress automatically? Do we really need to call a function to fill it?

Comment (john-hu) [inline tv_apps/tv-epg/js/epg.js]: I mean fill progress at the timing we set progress.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d74a7a2add70a3f821c251a27aaab3c2b7393fdd

**Pull request merged**